storage/Devices/DiskImpl.cc

```diff
@@ -107,19 +107,39 @@ namespace storage
     PtType
     Disk::Impl::get_default_partition_table_type() const
     {
-	const Arch& arch = get_devicegraph()->get_storage()->get_arch();
+	PtType ret = get_possible_partition_table_types().front();
+
+	y2mil("ret:" << toString(ret));
+
+	return ret;
+    }
+
 
-	PtType ret = PtType::MSDOS;
+    std::vector<PtType>
+    Disk::Impl::get_possible_partition_table_types() const
+    {
+	// TODO other archs
+
+	const Arch& arch = get_devicegraph()->get_storage()->get_arch();
 
 	unsigned long long int num_sectors = geometry.kbToSector(get_size_k());
-	y2mil("num_sectors:" << num_sectors);
+	bool size_ok_for_msdos = num_sectors <= (1ULL << 32) - 1;
```

Comment:
Is there no nice predefined constant for this? No MAX_UINT32 or something like that?